news/services/news.py

```diff
@@ -97,6 +110,90 @@ def create_news_service(data, reporter):
     return True, 'news created successfully'
 
 
+def update_news_service(data, news):
+    # check data to be valid
+    editable_fields = ['title', 'description', 'is_draft', 'categories']
+    if not all(key in editable_fields for key in data):
+        return False, 'bad data format'
+
+    # update categories
+    if 'categories' in data.keys():
+        categories = [get_category_id(category_name) for category_name in data.get('categories')]
+        data.pop('categories')
+        news.categories.set(categories)
+    # update other fields
+    for field, value in data.items():
+        # check field can be updated
+        setattr(news, field, value)
+    news.save()
+
+    # save changes into cache
+    resp = NewsSerializer(news).data
+    NewsCache.set_news_data(news.token, resp)
+    return True, 'news updated successfully'
+
+
+def delete_news(news):
```

Comment:
you should check that the reporter is the author of the news and then delete it

Reply:
i check it in views by using is_reporter_author function